python/tvm/ir/op.py

```diff
@@ -115,3 +127,43 @@ def _register(v):
         return v
 
     return _register(value) if value is not None else _register
+
+
+def register_op_intrin_lowering(
+    op_name,
+    target,
+    f=None,
+    level=10,
+    override=False,
+):
```

Comment:
oh BTW, the order of the arguments matter, and we need to double check if the following syntax works

```
@tvm.ir.op.register_op_intrin_lowering("tir.exp", target="cuda", override=True)
def my_exp(expr) -> expr:
    pass
```

Reply:
Thanks for point that out. I thought about this issue again and I think here `f` is used overwhelmingly more than `level` and `override`. In most cases, it's just `op_name`, `target` and `f`. Therefore, I insist that we keep the current order for most intuitive function usage.

Reply:
An alternative approach is to force kwargs, as demonstrated below, so that users are required to explicitly mark which argument is override and which is level

```python
def (a, b, *, c, d)
```

Reply:
In that case, it would be awkward to use the function as decorators as we would need to mark all the argument. What do you think?

Reply:
This is an interesting topic to discuss :-)

In our particular case, we could still provide default values for some arguments, like:

```python
>>> def f(a, *, b, c=1):
...   print(f"a = {a}, b = {b}, c = {c}")
...
>>> f(1, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: f() takes 1 positional argument but 2 were given
>>> f(1, b=2)
a = 1, b = 2, c = 1
```

Reply:
Ah I see. With default values things could be a lot easier. I would prefer to use `def register_intrin_lowering(op_name, target, *, f=None, level=10, override=False,)`